main.py

```diff
@@ -118,22 +113,32 @@ def setup_target_task_training(args, target_tasks, model, strict):
         # Look for target train checkpoints (available only if we're restoring from a run that
         # already finished), then look for training checkpoints.
 
-        best_path = get_best_checkpoint_path(args.run_dir)
+        best_path = get_best_checkpoint_path(args.run_dir, "target_train")
         if best_path:
             load_model_state(
                 model, best_path, args.cuda, task_names_to_avoid_loading, strict=strict
             )
         else:
-            assert_for_log(
-                args.allow_untrained_encoder_parameters, "No best checkpoint found to evaluate."
-            )
-
-            if args.transfer_paradigm == "finetune":
-                # Save model so we have a checkpoint to go back to after each
-                # task-specific finetune.
-                model_state = model.state_dict()
-                model_path = os.path.join(args.run_dir, "model_state_untrained_pre_target_train.th")
-                torch.save(model_state, model_path)
+            if args.do_pretrain == 1:
+                best_pretrain = get_best_checkpoint_path(args.run_dir, "pretrain")
+                if best_pretrain:
+                    load_model_state(
+                        model, best_pretrain, args.cuda, task_names_to_avoid_loading, strict=strict
+                    )
+                else:
+                    assert_for_log(
+                        args.allow_untrained_encoder_parameters,
```

Comment:
I think this check should occur _outside_ of the pretraining check, the logic being that if you're not doing pretraining, you are using an un(pre)trained model.

is the name "allow_untrained_encoder_parameters" accurate here? At the end of pretraining, even if you don't load any checkpoints, you still have trained parameters.

Reply:
This allows for random sentence encoders (skipping pretraining step to only run do_target_task_training). Moving this to if args.do_pretrianing = 0.